### PR TITLE
Letting it clear that the hosted zone should be public

### PR DESCRIPTION
**Pull request by caarlos0**
I was trying to use an already created hosted zone that was private. 
That doesn't work.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1548)

### CODE REVIEW

File: docs/aws.md [hunk 1]
```diff
@@ -90,7 +90,7 @@ We will now need to set up DNS for cluster, find one of the scenarios below (A,B
 
 If you bought your domain with AWS, then you should already have a hosted zone in Route53.
 
-If you plan on using your base domain, then no more work is needed. If you plan on using a subdomain to build your clusters on you will need to create a 2nd hosted zone in Route53.
+If you plan on using your base domain, then no more work is needed. If you plan on using a subdomain to build your clusters on you will need to create a 2nd hosted zone in Route53. Note that the hosted zone must be public and have a valid domain name.
```
Comment (kris-nova) [inline docs/aws.md]: Can we move this too the **Testing your DNS Setup** block? 

Also suggest wordsmithing:

> kops will asumme that the hosted zone you are using is publicly resolvable. If you plan on using a private DNS hosted zone, please set `--dns private`

Comment (caarlos0) [inline docs/aws.md]: way better, fixed

**Comment by caarlos0** [comment]
I signed it

**Comment by justinsb** [comment]
LGTM - thanks so much @caarlos0 !

**Comment by caarlos0** [comment]
Thanks!

**Pull request merged**